include/sdbus-c++/Message.h

```diff
@@ -85,10 +90,23 @@ namespace sdbus {
         Message& operator<<(const ObjectPath &item);
         Message& operator<<(const Signature &item);
         Message& operator<<(const UnixFd &item);
-        template <typename _Element> Message& operator<<(const std::vector<_Element>& items);
-        template <typename _Key, typename _Value> Message& operator<<(const std::map<_Key, _Value>& items);
-        template <typename... _ValueTypes> Message& operator<<(const Struct<_ValueTypes...>& item);
-        template <typename... _ValueTypes> Message& operator<<(const std::tuple<_ValueTypes...>& item);
+
+        template <typename _Element, typename _Allocator>
+        Message& operator<<(const std::vector<_Element, _Allocator>& items);
+        template <typename _Element, std::size_t _Size>
+        Message& operator<<(const std::array<_Element, _Size>& items);
+#if __cplusplus >= 202002L
+        template <typename _Element, std::size_t _Extent>
+        Message& operator<<(const std::span<_Element, _Extent>& items);
+#endif
+        template <typename _Key, typename _Value, typename _Compare, typename _Allocator>
+        Message& operator<<(const std::map<_Key, _Value, _Compare, _Allocator>& items);
+        template <typename _Key, typename _Value, typename _Hash, typename _KeyEqual, typename _Allocator>
+        Message& operator<<(const std::unordered_map<_Key, _Value, _Hash, _KeyEqual, _Allocator>& items);
+        template <typename... _ValueTypes>
+        Message& operator<<(const Struct<_ValueTypes...>& item);
+        template <typename... _ValueTypes>
+        Message& operator<<(const std::tuple<_ValueTypes...>& item);
```

Comment:
So, you do allow std::tuple? I thought the agreement was to not allow that and use `sdbus::Struct` instead.

Reply:
Yes, what you say holds. We use `Struct`. Nothing changed here regarding `operator<<(const std::tuple&)`, I just broke the declaration in two lines, hence it appears as a new stuff. The tuple operator there and needs to stay there, too, because it's used and needed by sdbus-c++ and its clients. It serves different purpose than `Struct`. See the a) and b) points in a comment of mine where I explained that: https://github.com/Kistler-Group/sdbus-cpp/issues/328#issuecomment-1651602916